socketio/src/socket.rs

```diff
@@ -41,8 +36,10 @@ pub struct Ack {
 
 /// Handles communication in the `socket.io` protocol.
 #[derive(Clone)]
-pub struct Socket {
-    engine_socket: Arc<EngineIoSocket>,
+pub struct Socket<
+    EngineSocket: rust_engineio::model::Socket + Send + Sync + Clone + 'static + Debug,
```

Comment:
Would it make sense to have those trait bounds directly on the Socket trait? I don't think there will be an implementation that's not Send + Sync + ...

Reply:
86d501ac1fcf7116630e3e8895b69616349eb89d